Honor advanced setting in templates defaults.yaml

DerAndereAndi opened the pull request:
`evcc configure` currently asks values for params that have `advanced: true` set in `defaults.yaml` even when it runs in `standard` mode where it shouldn't ask for any param defined as advanced. This PR fixes this.

Also simplify code that overwrites param values with defaults as discussed in https://github.com/evcc-io/evcc/pull/5824

DerAndereAndi committed:
Honor advanced setting in templates defaults.yaml

DerAndereAndi committed:
Update docs

andig commented:
> This PR fixes this.

@DerAndereAndi meaning it won't ask for those questions?

DerAndereAndi committed:
Simplify param merging

- Convert `bool` fields in `Param` to `*bool`
- Use merge() instead of manual field merges

DerAndereAndi commented:
@andig correct

Now also added your suggestion for simplification of overwriting field values

andig commented:
Sehr cool. Ich hätte mich mit dem Thema so schnell nicht wieder auseinander gesetzt...
